Automattic/jetpack
# Improve My Jetpack checkout flow consistency

## Proposed changes:
This PR includes several changes to the product purchase and "start for free" flows in My Jetpack

* Products now all default to installing the standalone plugin if one is available
* All products will go to their individual product page after a purchase, if there is one available
* Bug fixed on the product table views where there was no "loading" state after clicking on start for free
* Enables the "connect after checkout" flow for free purchases that need to go through the cart (Search free)
* Fixes inconsistent post-checkout redirect behavior that was happening between product card and product table interstitial screens
* Fixes bug in free search product activation where errors were returned for no plan as the user was trying to get a free plan

## Does this pull request change what data or activity we track or use?


* No

## Testing instructions:






* It is best to test this PR on a fresh Jetpack install with no standalone plugins installed - the local install already has copies of all plugins installed
* Navigate to My Jetpack, test purchasing different products from My Jetpack. Note that this PR expects that all purchases are started from the My Jetpack screen by clicking on one of the "Learn More" buttons on the product cards.
* Test the "start for free" options for different product (example below).
* ![Screenshot 2024-02-23 at 11 51 12 AM](https://github.com/Automattic/jetpack/assets/18016357/1a7bcd54-9c4f-481b-a4f6-7834eda69e8d)
* Try removing all purchases, deleting all plugins and disconnecting a few times to see what happens when you purchase disconnected or make purchases without plugins installed.

**Expected behaviors while testing**
*  When purchasing a product from My Jetpack (whenever you go through the checkout) - if that product has an available standalone plugin, the standalone should be installed and activated. After the purchase, you should be redirected to the plugin page in wp-admin.
* When choosing a "start for free" option, the standalone plugin should be installed/ activated and you should be redirected to the plugin page.

**Known Issues**
* If your very first "purchase" while Jetpack is disconnected is Jetpack Search Free - you will encounter an error during the checkout process. This is being addressed in this PR: https://github.com/Automattic/wp-calypso/pull/87958
* For initial disconnected purchases, some products will link to the cloud from the license assignment thank you screen (screenshot below). This happens for backup, scan and search. This is out of scope for this PR

![Screenshot 2024-02-27 at 2 55 13 PM](https://github.com/Automattic/jetpack/assets/18016357/523c4956-c262-4c27-9f1b-8dbd98330bfe)

* On the screen pictured above, some products might link to the incorrect page in wp-admin (ex: you might buy Boost and get linked to the Backups plugin page in wp-admin), this is being addressed in this PR: https://github.com/Automattic/wp-calypso/pull/87961
